### PR TITLE
Replace context keys by constants

### PR DESCRIPTION
**Pull request by soyuka**
Rebases #1407

### CODE REVIEW

**Comment by meyerbaptiste** [comment]
You can do the same for all the other constants (`ATTRIBUTES`, `ALLOW_EXTRA_ATTRIBUTES`, `GROUPS`, etc.)!

**Comment by soyuka** [comment]
@meyerbaptiste this is done btw

**Comment by meyerbaptiste** [comment]
Huum, I think you forget one `enable_max_depth` (`AbstractObjectNormalizer::ENABLE_MAX_DEPTH`) in the `EagerLoadingExtension` class!

**Comment by meyerbaptiste** [comment]
And what about the `tests` directory? :sweat_smile:

File: src/Serializer/Filter/GroupFilter.php [hunk 1]
```diff
@@ -57,7 +58,7 @@ public function apply(Request $request, bool $normalization, array $attributes,
     public function getDescription(string $resourceClass): array
     {
         return [
-            $this->parameterName.'[]' => [
+            "{$this->parameterName}[]" => [
```
Comment (sroze) [inline src/Serializer/Filter/GroupFilter.php]: This looks less clear to me

Comment (soyuka) [reply src/Serializer/Filter/GroupFilter.php]: Dunno why I changed this actually. Reverted.

**Comment by dunglas** [comment]
Thanks @tuanphpvn and @soyuka!

**Pull request merged**